addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/serialization/ListUiActionSerializer.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.neeo.internal.serialization;
+
+import java.lang.reflect.Type;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.io.neeo.internal.models.ListUiAction;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+/**
+ * Implementation of {@link JsonSerializer} and {@link JsonDeserializer} to serialize/deserialize
+ * {@link ListUiAction}
+ *
+ * @author Tim Roberts - Initial Contribution
+ */
+@NonNullByDefault
+public class ListUiActionSerializer implements JsonSerializer<ListUiAction>, JsonDeserializer<ListUiAction> {
+    @Override
+    public ListUiAction deserialize(@Nullable JsonElement elm, @Nullable Type type,
+            @Nullable JsonDeserializationContext jsonContext) throws JsonParseException {
+        Objects.requireNonNull(elm, "elm cannot be null");
```

Comment:
The gson.JsonSerializer and gson.JsonDeserializer interfaces require nullable attribute (I get the "illegal redefinition of parameter" when I don't have the nullable attribute).  Since it's marked as nullable, unless i check for requireNotNull - I get the "potential for null" warnings when I try to use it.  So basically - it has to be Nullable and I need to ensure it's not null

Reply:
Thanks!